concordium-consensus/src/Concordium/Queries.hs

```diff
@@ -1423,3 +1425,108 @@ getNumberOfNonFinalizedTransactions =
+    getBakersConsensusV0 :: (SkovQueryMonad m, PVSupportsDelegation (MPV m)) => BlockState m -> m [BakerRewardPeriodInfo]
+    getBakersConsensusV0 bs = do
+        bakers <- BS.getCurrentEpochBakers bs
+        finalizationParameters <- genesisFinalizationParameters . _gcCore <$> getGenesisData
+        totalCCD <- rsTotalAmount <$> BS.getRewardStatus bs
+        let finalizationCommittee = makeFinalizationCommittee finalizationParameters totalCCD bakers
+        mapBakersToInfos bs (Vec.toList $ fullBakerInfos bakers) (partyBakerId <$> Vec.toList (parties finalizationCommittee))
+    -- Get the bakers and calculate the finalization committee for protocols using consensus v1.
+    getBakersConsensusV1 :: (BS.BlockStateQuery m, IsConsensusV1 (MPV m)) => BlockState m -> m [BakerRewardPeriodInfo]
+    getBakersConsensusV1 bs = do
+        bakers <- BS.getCurrentEpochBakers bs
+        finCommitteeParams <- BS.getCurrentEpochFinalizationCommitteeParameters bs
+        let finalizationCommittee = ConsensusV1.computeFinalizationCommittee bakers finCommitteeParams
+        mapBakersToInfos bs (Vec.toList $ fullBakerInfos bakers) (SkovV1.finalizerBakerId <$> Vec.toList (SkovV1.committeeFinalizers finalizationCommittee))
+    -- Map bakers to their assoicated 'BakerRewardPeriodInfo'.
+    -- The supplied bakers and list of baker ids (of the finalization committee) MUST
+    -- be sorted in ascending order of their baker id.
+    mapBakersToInfos ::
+        ( BS.BlockStateQuery m,
+          PVSupportsDelegation (MPV m)
+        ) =>
+        -- The block state to request the pool status from.
+        BlockState m ->
+        -- All bakers for the reward period.
+        [FullBakerInfo] ->
+        -- The baker ids of the finalizers for the reward period.
+        [BakerId] ->
+        m [BakerRewardPeriodInfo]
+    mapBakersToInfos bs fullBakerInfos finalizersByBakerId = fst <$> foldM mapBaker ([], finalizersByBakerId) fullBakerInfos
+      where
+        -- No finalizers left to pick off, so this baker must only be
+        -- member of the baking committee.
+        mapBaker (acc, []) baker = do
+            info <- toBakerRewardPeriodInfo False bs baker
+            return (info : acc, [])
+        -- Check whether the baker id of the finalizer candidate matches the
+        -- bakers id. If this is the case then the baker is member of the finalization committee and
+        -- otherwise not.
+        mapBaker (acc, finalizers@(candidate : remaining)) baker = do
+            let isFinalizer = (baker ^. theBakerInfo . to _bakerIdentity) == candidate
+            if isFinalizer
+                then do
+                    info <- toBakerRewardPeriodInfo True bs baker
+                    return (info : acc, remaining)
+                else do
+                    info <- toBakerRewardPeriodInfo False bs baker
+                    return (info : acc, finalizers)
```

Comment:
This will reverse the two lists. It'd be better to return them in the order, but I guess we don't promise that in the API.

Reply:
Yes. I did think about this but convinced myself that we do not promise anything. But I think you're right - they should be delivered in ascending order of bakerid.

Reply:
Fixed.